### PR TITLE
Attachments Feature Flag

### PR DESCRIPTION
**Pull request by dsainati1**
Closes https://github.com/onflow/cadence/issues/2354

To allow us to merge the feature branch to master, put any uses of the attachments features (attachment declarations, attach expressions, remove declarations, and type indexing expressions) behind a feature flag, which will throw an error in the checker if not enabled. 
______

- [ ] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

**Review by SupunS — APPROVED** [review]
Nice!

File: runtime/sema/errors.go [hunk 1]
```diff
@@ -4001,3 +4001,19 @@ func (e *InvalidTypeIndexingError) Error() string {
...
+
+var _ SemanticError = &AttachmentsNotEnabledError{}
+var _ errors.UserError = &AttachmentsNotEnabledError{}
+
+func (*AttachmentsNotEnabledError) isSemanticError() {}
+
+func (*AttachmentsNotEnabledError) IsUserError() {}
+
+func (e *AttachmentsNotEnabledError) Error() string {
+	return "the attachments feature is not yet enabled and cannot be used"
```
Comment (turbolent) [inline runtime/sema/errors.go:4018]:
```suggestion
	return "attachments are not enabled and cannot be used in this environment"
```

**Review by turbolent — COMMENTED** [review]
Nice!

**Comment by turbolent** [comment]
GitHub still has this bug where changing the target branch causes CI to not kick off. Closed and re-opened to trigger CI

**Review by turbolent — APPROVED** [review]
Nice!

**Pull request merged**